File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input = sys.stdin.readline

n = int(input())
n_num = list(map(int, input().split()))
m = int(input())
m_num = list(map(int, input().split()))

n_num.sort()

def find_num(num):
    left = 0
    right = len(n_num) - 1
    ischecked = 0
    while left <= right:
        mid = (left + right) // 2
        if n_num[mid] < num:
            left = mid + 1
        elif n_num[mid] > num:
            right = mid - 1
        else:
            ischecked = 1
            break
    return ischecked

for num in m_num:
    print(find_num(num))

File: 백준/Silver/1920. 수 찾기/수 찾기.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().trim().split('\n')

const N = Number(input[0])
const arr = input[1].split(' ').map(Number)
arr.sort((a, b) => a - b)

const M = Number(input[2])
const nums = input[3].split(' ').map(Number)

const answer = []
for (const num of nums) {
  answer.push(find_num(num))
}
console.log(answer.join('\n'))

function find_num(num) {
  let left = 0
  let right = N - 1
  while(left <= right) {
    const mid = Math.floor((left + right) / 2)

    if (arr[mid] < num) {
      left = mid + 1
    } else if (num < arr[mid]) {
      right = mid - 1
    } else return 1
  }

  return 0
}